src/hotspot/share/opto/library_call.cpp

```diff
@@ -3172,13 +3173,15 @@ bool LibraryCallKit::inline_native_jvm_commit() {
   Node* next_pos_X = _gvn.transform(ConvL2X(arg));
 
   // Store the next_position to the underlying jfr java buffer.
-  Node* commit_memory;
 #ifdef _LP64
-  commit_memory = store_to_memory(control(), java_buffer_pos_offset, next_pos_X, T_LONG, MemNode::release);
+  store_to_memory(control(), java_buffer_pos_offset, next_pos_X, T_LONG, MemNode::release);
 #else
-  commit_memory = store_to_memory(control(), java_buffer_pos_offset, next_pos_X, T_INT, MemNode::release);
+  store_to_memory(control(), java_buffer_pos_offset, next_pos_X, T_INT, MemNode::release);
```

Comment:
BTW, you can write this is less duplication, like:

```
store_to_memory(control(), java_buffer_pos_offset, next_pos_X, LP64_ONLY(T_LONG) NOT_LP64(T_INT), MemNode::release);
```

(I wish there was a `X`-style macro like `MakeConX`, but I don't think there is one for `BasicType`)

Reply:
Nice!